x-pack/plugin/logstash/src/main/java/org/elasticsearch/xpack/logstash/action/TransportGetPipelineAction.java

```diff
@@ -122,37 +141,66 @@ private void handleFailure(Exception e, ActionListener<GetPipelineResponse> list
         }
     }
 
-    private void handleSearchResponse(
+    private void handleFilteringSearchResponse(
         SearchResponse searchResponse,
         Map<String, BytesReference> pipelineSources,
+        Set<String> explicitPipelineIds,
+        Set<String> wildcardPipelineIdPatterns,
+        int maxNumHitsToFetch,
         Consumer<SearchResponse> clearScroll,
         ActionListener<GetPipelineResponse> listener
     ) {
+        maxNumHitsToFetch += searchResponse.getHits().getHits().length;
```

Comment:
Since we assign a new variable later to hold this value (`finalNumHitsToFetch`) can we instead make a new variable here to hold this mutation instead of overwriting the parameter passed in?